### ISSUE TITLE
How to apply spin polarized current for STT induced dynamics calculations?

### ISSUE THREAD

**Issue by karpathyan** [richard-evans/vampire#109]
Hi, 
              I'm trying to do a simulation where we try to move the Skyrmions and domain walls with a spin-polarized current via STT. How can I apply a spin-current? I have seen papers where they did the same thing with VAMPIRE, I looked in the manual and also in the workshop examples. I could not find anything. 
              
      Thanks for any assistance!

**Comment by xbunax** [comment]
I have submit a [pr](https://github.com/richard-evans/vampire/pull/108#issue-2231398806), which can control the electrical strength and shape.

**Comment by karpathyan** [comment]
> I have submit a [pr](https://github.com/richard-evans/vampire/pull/108#issue-2231398806), which can control the electrical strength and shape.

Hi, thank you for the response. I looked at your PR and it applies an electric field. What want is a **spin polarized** electric current.

**Comment by xbunax** [comment]
> > I have submit a [pr](https://github.com/richard-evans/vampire/pull/108#issue-2231398806), which can control the electrical strength and shape.
> 
> Hi, thank you for the response. I looked at your PR and it applies an electric field. What want is a **spin polarized** electric current.

yes, the stt is induced by current. So which parameter do you want to change. In VAMPIRE, the stt field is applied by `spin-transfer-torque` which contents $a_J$ and `spin-transfer-torque-polarization-unit-vector`

**Comment by karpathyan** [comment]
> > > I have submit a [pr](https://github.com/richard-evans/vampire/pull/108#issue-2231398806), which can control the electrical strength and shape.
> > 
> > 
> > Hi, thank you for the response. I looked at your PR and it applies an electric field. What want is a **spin polarized** electric current.
> 
> yes, the stt is induced by current. So which parameter do you want to change. In VAMPIRE, the stt field is applied by `spin-transfer-torque` which contents aJ and `spin-transfer-torque-polarization-unit-vector`

Thank you, can you please share an example simulation if you have one? Because I could not find the options for `spin-transfer-torque` in the Vampire tutorial.

**Comment by xbunax** [comment]
> > > > I have submit a [pr](https://github.com/richard-evans/vampire/pull/108#issue-2231398806), which can control the electrical strength and shape.
> > > 
> > > 
> > > Hi, thank you for the response. I looked at your PR and it applies an electric field. What want is a **spin polarized** electric current.
> > 
> > 
> > yes, the stt is induced by current. So which parameter do you want to change. In VAMPIRE, the stt field is applied by `spin-transfer-torque` which contents aJ and `spin-transfer-torque-polarization-unit-vector`
> 
> Thank you, can you please share an example simulation if you have one? Because I could not find the options for `spin-transfer-torque` in the Vampire tutorial.

you can git clone and switch to develop branch，use `ripgrep` to grep this flag if you have
